File: src/app/Class/GenreInstrument.ts
import { Genre } from "./Genre";
import { Instrument } from "./Instrument";

export class GenreInstrument {
  private id: number;
  private genre: Genre;
  private instrument: Instrument;

  constructor(obj?: any) {
    this.id = (obj && obj.id) || 0;
    this.genre = (obj && obj.genre) || null;
    this.instrument = (obj && obj.instrument) || null;
  }
}

File: src/main/java/MusicCube/entities/GenreInstrument.java
package MusicCube.entities;

import javax.persistence.*;

@Entity
public class GenreInstrument {

    @Id
    @GeneratedValue
    @Column
    private int id;

    @ManyToOne
    private Genre genre;

    @ManyToOne
    private Instrument instrument;

    public GenreInstrument() {}

    public GenreInstrument(Genre genre, Instrument instrument) {
        this.genre = genre;
        this.instrument = instrument;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public Genre getGenre() {
        return genre;
    }

    public void setGenre(Genre genre) {
        this.genre = genre;
    }

    public Instrument getInstrument() {
        return instrument;
    }

    public void setInstrument(Instrument instrument) {
        this.instrument = instrument;
    }
}
